src/Compilers/CSharp/Portable/Binder/Binder.ValueChecks.cs

```diff
@@ -523,6 +523,31 @@ private BoundExpression CheckValue(BoundExpression expr, BindValueKind valueKind
                     Debug.Assert(valueKind is (BindValueKind.Assignable or BindValueKind.RefOrOut or BindValueKind.RefAssignable) || diagnostics.DiagnosticBag is null || diagnostics.HasAnyResolvedErrors());
                     return expr;
 
+                case BoundKind.PropertyAccess:
+                    if (!InAttributeArgument)
+                    {
+                        var propertyAccess = (BoundPropertyAccess)expr;
+                        if (propertyAccess.PropertySymbol.RefKind == RefKind.None)
+                        {
+                            bool? useAsLvalue = RequiresRValueOnly(valueKind) ? false : RequiresAssignableVariable(valueKind) ? true : null;
+                            if (useAsLvalue.HasValue)
+                            {
+                                if (CanUseBackingFieldDirectlyInConstructor(propertyAccess, useAsLvalue.Value))
+                                {
+                                    // Update the property access to indicate that the backing field should be used directly.
+                                    expr = propertyAccess.Update(
+                                        propertyAccess.ReceiverOpt,
+                                        propertyAccess.InitialBindingReceiverIsSubjectToCloning,
+                                        propertyAccess.PropertySymbol,
+                                        useAsLvalue.Value ? AccessorKind.Set : AccessorKind.Get, // PROTOTYPE: What about compound assignment, increment?
```

Comment:
Consider extracting and reusing logic from 'GetIndexerAccessorKind' in order to determine this.